src/parse.rs

```diff
@@ -160,11 +170,78 @@ pub fn get_mount_from<T, U>(target: T, iter: MountIter<U>)
     Ok(ret)
 }
 
+pub fn get_mount_search_from<U>(search: &Search, iter: MountIter<U>)
+        -> Result<Option<Vec<MountEntry>>, ParseError> where U: BufRead {
+    let mut ret = Vec::new();
+    for mount in iter {
+        match mount {
+            Ok(m) => {
+                match search {
```

Comment:
This seems wordy. Why not match on the search and then something like:

```iter().filter(<function that keeps error and all matches>)```

something like:

```|r| r.is_err() || r.spec == spec```

and so forth.

Or something, depending on your attitude to errors.